server/src/test/java/com/paypal/selion/grid/servlets/GridStatisticsTest.java

```diff
@@ -0,0 +1,135 @@
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.web.servlet.RegistryBasedServlet;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.paypal.selion.pojos.BrowserInformationCache;
+import com.paypal.selion.pojos.BrowserStatisticsCollection;
+import com.paypal.selion.pojos.BrowserStatisticsCollection.BrowserStatistics;
+
+/**
+ * Tests for GridStatistics servlet
+ */
+@PrepareForTest({ GridStatistics.class, BrowserInformationCache.class, RegistryBasedServlet.class, Registry.class })
+public class GridStatisticsTest extends PowerMockTestCase {
+
+    @BeforeClass
+    public void beforeClass() {
+
+    }
+
+    @Test
+    public void testValidRequest() throws Exception {
+        HttpServletRequest request = PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(request.getHeader("Accept")).thenReturn("*/*");
+        HttpServletResponse response = PowerMockito.mock(HttpServletResponse.class);
+        ServletOutputStream servletOutputStream = new ServletOutputStream() {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+            @Override
+            public void write(int b) throws IOException {
+                bos.write(b);
+            }
+
+            @Override
+            public void setWriteListener(WriteListener arg0) {
+            }
+
+            @Override
+            public boolean isReady() {
+                return true;
+            }
+
+            public String toString() {
+                return bos.toString();
+            }
+        };
+
+        PowerMockito.when(response.getOutputStream()).thenReturn(servletOutputStream);
+        List<BrowserStatistics> browserStatisticsList = new ArrayList<BrowserStatistics>();
+        BrowserStatisticsCollection t = new BrowserStatisticsCollection();
+        BrowserStatistics browserStatistics = t.new BrowserStatistics("chrome");
+        browserStatisticsList.add(browserStatistics);
+        spy(GridStatistics.class);
+        stub(method(GridStatistics.class, "getGridLoadResponse")).toReturn(browserStatisticsList);
+        GridStatistics gridStatistics = new GridStatistics();
+        gridStatistics.doPost(request, response);
+        Assert.assertEquals(servletOutputStream.toString(),
+                "[{\"browserName\":\"chrome\",\"statistics\":{\"waitingRequests\":0,\"maxBrowserInstances\":0}}]",
+                "Servlet output not matching");
+
+    }
+
+    @Test
+    public void testEmptyRequest() throws Exception {
+        HttpServletRequest request = PowerMockito.mock(HttpServletRequest.class);
```

Comment:
The project has a dependency on spring-test now which includes MockHttpServletRequest & MockHttpServletResponse objects.. Can you use that instead here. This would reduce some of the code below.